File: algorithms/easy/most_frequent_even_element.cpp
class Solution {
public:
  int mostFrequentEven(vector<int>& nums) {
    unordered_map<int, int> mp;
    int ans = -1;
    int max_freq = -1;
    for (auto n : nums) {
      if (n % 2 == 1) continue;
      mp[n]++;
      if (mp[n] > max_freq) {
        max_freq = mp[n];
        ans = n;
      } else if (mp[n] == max_freq) {
        ans = min(n, ans);
      }
    }
    return ans;
  }
};

File: algorithms/easy/most_frequent_even_element.ex
defmodule Solution do
  @spec most_frequent_even(nums :: [integer]) :: integer
  def most_frequent_even(nums) do
    nums
    |> Enum.reduce(%{}, fn n, acc ->
      if rem(n, 2) == 0 do
        Map.update(acc, n, 1, &(&1 + 1))
      else
        acc
      end
    end)
    |> Enum.max_by(fn {k, v} -> {v, -k} end, fn -> {-1} end)
    |> elem(0)
  end
end
